project/Build.scala

```diff
@@ -1088,84 +1088,101 @@ object Build extends sbt.Build {
-
-          jsDependencies += ProvidedJS / "ScalaJSDefinedTestNatives.js" % "test",
+  def testSuiteCommonSettings(addShared: Boolean): Seq[Setting[_]] = Seq(
+    publishArtifact in Compile := false,
+    scalacOptions ~= (_.filter(_ != "-deprecation")),
 
-          scalaJSSemantics ~= (_.withRuntimeClassName(_.fullName match {
-            case "org.scalajs.testsuite.compiler.ReflectionTest$RenamedTestClass" =>
-              "renamed.test.Class"
-            case fullName =>
-              fullName
-          })),
+    // Need reflect for typechecking macros
+    libraryDependencies +=
+      "org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided",
 
-          sources in Test ++= {
-            def listAllScalaFilesIf(testDir: String, condition: Boolean): Seq[File] =
-              if (condition) (((sourceDirectory in Test).value / testDir) ** "*.scala").get
-              else Nil
+    testOptions += Tests.Argument(TestFrameworks.JUnit, "-v", "-a"),
 
-            val hasSAM = {
-              scalaBinaryVersion.value != "2.10" &&
-              scalacOptions.value.contains("-Xexperimental")
-            }
-
-            listAllScalaFilesIf("require-jdk7", javaVersion.value >= 7) ++
-            listAllScalaFilesIf("require-jdk8", javaVersion.value >= 8) ++
-            listAllScalaFilesIf("require-sam", hasSAM)
-          },
-
-          /* Generate a scala source file that throws exceptions in
-             various places (while attaching the source line to the
-             exception). When we catch the exception, we can then
-             compare the attached source line and the source line
-             calculated via the source maps.
-
-             see test-suite/src/test/resources/SourceMapTestTemplate.scala
-           */
-          sourceGenerators in Test <+= Def.task {
-            val dir = (sourceManaged in Test).value
-            IO.createDirectory(dir)
-
-            val template = IO.read((resourceDirectory in Test).value /
-              "SourceMapTestTemplate.scala")
-
-            def lineNo(cs: CharSequence) =
-              (0 until cs.length).count(i => cs.charAt(i) == '\n') + 1
-
-            var i = 0
-            val pat = "/\\*{2,3}/".r
-            val replaced = pat.replaceAllIn(template, { mat =>
-              val lNo = lineNo(mat.before)
-              val res =
-                if (mat.end - mat.start == 5)
-                  // matching a /***/
-                  s"if (TC.is($i)) { throw new TestException($lNo) } else "
-                else
-                  // matching a /**/
-                  s"; if (TC.is($i)) { throw new TestException($lNo) } ;"
+    sources in Test ++= {
+      def listScalaFilesIf(testDir: File, condition: Boolean): Seq[File] =
+        if (condition) (testDir ** "*.scala").get
+        else Nil
 
-              i += 1
+      val hasSAM = {
+        scalaBinaryVersion.value != "2.10" &&
+          scalacOptions.value.contains("-Xexperimental")
```

Comment:
Indentation issue (the second line should not be more indented that the first line of the condition)